packages/lab/src/pill-next/PillNext.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  forwardRef,
+  MouseEvent,
+  PropsWithChildren,
+  KeyboardEvent,
+  ComponentPropsWithoutRef,
+} from "react";
+import clsx from "clsx";
+import { useWindow } from "@salt-ds/window";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { makePrefixer, useButton } from "@salt-ds/core";
+import pillCss from "./PillNext.css";
+
+export type PillClickEvent =
+  | MouseEvent<Element, globalThis.MouseEvent>
+  | KeyboardEvent<HTMLDivElement>;
+
+export interface PillNextProps extends ComponentPropsWithoutRef<"div"> {
+  /* If true the pill will be disabled */
+  disabled?: boolean;
+  /* Pass an element to render an icon descriptor on the left of the label */
+  icon?: React.ReactNode;
+  /* Pass a callback function to make the pill clickable */
+  onClick?: (event: PillClickEvent) => void;
+}
+
+const withBaseName = makePrefixer("saltPillNext");
+
+export const PillNext = forwardRef<
+  HTMLDivElement,
+  PropsWithChildren<PillNextProps>
```

Comment:
We don't need the PropsWithChildren since children is included in ComponentPropsWithoutRef